src/org/labkey/targetedms/SkylineAuditLogManager.java

```diff
@@ -541,21 +560,21 @@ public void addAVersionTest() throws IOException, AuditLogException
             }
 
             assertNotNull(tree);
-            assertEquals(14, tree.getTreeSize());
+            assertEquals(15, tree.getTreeSize());
```

Comment:
This seems odd to me. Why are we getting additional entries in the tree? Perhaps we need to filter on the run ID to keep things consistent? I worry because places like SkylineAuditLogManager.deleteDocumentVersionLog() deletes the entries returned by buildLogTree() and we don't want deleting one document to mess up a separate copy of that document in a different folder.

Reply:
So the audit log tree now contains a node for every auditLogEntry/versionId combo.  This is necessary for adding the junction table, and it's exactly why SkylineAuditLogManager.deleteDocumentVersionLog does not delete auditLogEntries that belong to multiple runs.  This jest test contains many duplicate audit log entries across the runs. Only the leaf nodes of the audit log entry tree have the versionId so that is where new nodes are showing up in the tree.  That being said, I have somewhat limited testing data and scenarios here so happy to do some side by side testing on PanoramaWeb.org or wherever else there may be references.